spec/rubocop/cop/style/redundant_ternary_spec.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+describe RuboCop::Cop::Style::RedundantTernary do
+  let(:config) { RuboCop::Config.new }
+  subject(:cop) { described_class.new(config) }
+
+  before { inspect_source(cop, source) }
+
+  shared_examples 'code with offense' do |code, expected|
+    context "when checking #{code}" do
+      let(:source) { code }
+
+      it 'registers an offense' do
+        expect(cop.offenses.size).to eq(1)
+        expect(cop.messages).to eq(['Returning true/false from a conditional is unnecessary'])
+      end
+
+      if expected
+        it 'auto-corrects' do
+          expect(autocorrect_source(cop, code)).to eq(expected)
+        end
+
+        it 'claims to auto-correct' do
+          autocorrect_source(cop, code)
+          expect(cop.offenses.last.status).to eq(:corrected)
+        end
+      else
+        it 'does not auto-correct' do
+          expect(autocorrect_source(cop, code)).to eq(code)
+        end
+
+        it 'does not claim to auto-correct' do
+          autocorrect_source(cop, code)
+          expect(cop.offenses.last.status).to eq(:uncorrected)
+        end
+      end
+    end
+  end
+
+  shared_examples 'code without offense' do |code|
+    let(:source) { code }
+
+    it 'does not register an offense' do
+      expect(cop.offenses).to be_empty
+    end
+  end
+
+  it_behaves_like "code with offense",
+                  "1 == 2 ? true : false",
+                  "1 == 2"
+
+  it_behaves_like "code with offense",
+                  "x == y ? true : false",
+                  "x == y"
+
+  it_behaves_like "code with offense",
+                  "x == y ? false : true",
+                  "!(x == y)"
+
+  it_behaves_like "code without offense",
+                  "x == y ? 1 : 10"
+
+  xcontext "TODO" do
```

Comment:
We don't use code for project tracking. Leave a GitHub issue if you intend to do something later. 🙂